--- ftplugin/tex_nine/__init__.py	2015-01-15 13:46:23.019961908 +0000
+++ ftplugin/tex_nine/__init__.py	2015-01-15 13:47:31.291082108 +0000
@@ -282,7 +282,7 @@
                 e = messages['MASTER_NOT_ACTIVE'].format(path.basename(master))
                 raise TeXNineError(e)
             else:
-                match = re.search(r'\\bibliography{([^}]+)}',
+                match = re.search(r'\\(?:bibliography|addbibresource){([^}]+)}',
                                   masterbuffer)
                 if not match:
                     e = messages['NO_BIBTEX']

@@ -292,7 +292,8 @@ class TeXNineBibTeX(TeXNineBase):
                 dirname = path.dirname(master)
                 # Find the absolute paths of the bibfiles
                 for b in bibfiles:
-                    b += '.bib'
+                    if not b.endswith('.bib'):
+                        b += '.bib'
                     # Check if the bibfile is in the compilation folder
                     bibtemp = path.join(dirname, b)
                     b = ( bibtemp if path.exists(bibtemp) else b )
@@ -347,7 +347,7 @@
 
     @TeXNineBase.multi_file
     def _labels(self, vimbuffer,
-                pat=re.compile(r'\\label{(?P<label>[^,}]+)}|\\in(?:clude|put){(?P<fname>[^}]+)}')):
+                pat=re.compile(r'\\label{(?P<label>[^,}]+)}|\\in(?:clude\*|put){(?P<fname>[^}]+)}')):
         """Labels for references.
 
         Searches \label{} statements in the master file and in
